### PR TITLE
IBX-3582: Extend catalog-related PHP API doc

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-3582
| Versions      | 4.2+

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: code_samples/api/product_catalog/src/Command/CatalogCommand.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types=1);
```
Comment (adamwojs) [inline code_samples/api/product_catalog/src/Command/CatalogCommand.php:1]:
```suggestion
<?php 

declare(strict_types=1);
```

Please apply this suggestion to all commands.

File: code_samples/api/product_catalog/src/Command/CatalogCommand.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+{
+    private UserService $userService;
+
+    private PermissionResolver $permissionResolver;
+
+    private ProductServiceInterface $productService;
+
+    private CatalogServiceInterface $catalogService;
+
+    public function __construct(UserService $userService, PermissionResolver $permissionResolver, ProductServiceInterface $productService, CatalogServiceInterface $catalogService)
```
Comment (adamwojs) [inline code_samples/api/product_catalog/src/Command/CatalogCommand.php:29]: You could improve readability by putting each parameter in separate line

Please apply this suggestion to all commands.

File: code_samples/api/product_catalog/src/Command/ProductVariantCommand.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+final class ProductVariantCommand extends Command
+{
+    private $userService;
+
+    private $permissionResolver;
+
+    private $productService;
+
+    private $localProductService;
```
Comment (adamwojs) [inline code_samples/api/product_catalog/src/Command/ProductVariantCommand.php:24]: Property type is missing

File: docs/pim/catalog_api.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+To get a single catalog, you can use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:
+
+``` php
+[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 70, 72) =]]
+```
+
+## Get products in catalog
+
+To get products from a catalog, request the product query from the catalog object  with `catalog::getQuery()`.
```
Comment (adamwojs) [inline docs/pim/catalog_api.md:19]: Did you mean `\Ibexa\Contracts\ProductCatalog\Values\CatalogInterface::getQuery` ?

File: docs/pim/catalog_api.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+description: Use PHP API to get information about and manage product catalogs.
+---
+
+# Catalog API
+
+To get information about product catalogs and manage them, use `CatalogServiceInterface`.
+
+## Get catalog
+
+To get a single catalog, you can use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:
```
Comment (MagdalenaZuba) [inline docs/pim/catalog_api.md:11]:
```suggestion
To get a single catalog, use `CatalogServiceInterface::getCatalog()` and provide it with catalog ID, or `CatalogServiceInterface::getCatalogByIdentifier()` and pass the identifier:
```

File: docs/pim/catalog_api.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+with `Ibexa\Contracts\ProductCatalog\Values\CatalogInterface::getQuery()`.
+Then, create a new `ProductQuery` based on it and run a product search with `ProductServiceInterface::findProduct()`:
+
+``` php
+[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 80, 86) =]]
+```
+
+## Create catalog
+
+To create a catalog, you need to prepare a `CatalogCreateStruct`, with the identifier, name and description and Criteria for filtering products.
```
Comment (MagdalenaZuba) [inline docs/pim/catalog_api.md:29]:
```suggestion
To create a catalog, you need to prepare a `CatalogCreateStruct` that contains: identifier, name, description and Criteria for filtering products.
```

File: docs/pim/catalog_api.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+Then, pass this struct to `CatalogServiceInterface::createCatalog()`:
+
+``` php
+[[= include_file('code_samples/api/product_catalog/src/Command/CatalogCommand.php', 66, 74) =]]
+```
+
+## Update catalog
+
+You can update an existing catalog by using `CatalogServiceInterface::updateCatalog()`,
+to which you must pass the catalog object and a `CatalogUpdateStruct`.
```
Comment (MagdalenaZuba) [inline docs/pim/catalog_api.md:39]:
```suggestion
Use `CatalogServiceInterface::updateCatalog()` to update an existing catalog.
You must pass the catalog object and a `CatalogUpdateStruct` to the interface.
```

**Review by MagdalenaZuba — APPROVED** [review]

File: docs/pim/catalog_api.md (hunk 7 above)
Comment (DominikaK) [inline docs/pim/catalog_api.md:39]:
```suggestion
Use `CatalogServiceInterface::updateCatalog()` to update an existing catalog.
You must pass the catalog object and a `CatalogUpdateStruct` to the method.
```

**Pull request merged**